docs/src/contribute/design/next-tedge-api/registration.md

```diff
@@ -0,0 +1,227 @@
+---
+title: Registration
+tags: [Documentation]
+sidebar_position: 3
+---
+
+## Overview
+
+Automatic registration relies on user-defined configuration to provide a link between the topic names and the entity/component types.
```

Comment:
Rather that starting with *automatic* vs *user-defined* registration I would start with *registration*. 1) what's the point? 2) This can be done automatically when following naming conventions 3) this can be tuned to any use case.